### DISCUSSION TITLE
Shader reloading and console in Malt 1.0

### DISCUSSION THREAD

**Discussion by panthavma** [bnpr/Malt#267] [Q&A] [answered]
Hello! This might be a dumb question, but I don't seem to be able to find how to reload shaders directly from Blender to see changes. I am currently using the MiniPipeline example and a file containing the default shader defined in the same pipeline.

I have loaded my .glsl file in a material, but I don't seem to be able to find how to recompile it. Trying to reload the file, switch to another file and then to the original one again, or reloading the pipeline don't seem to work.

As an additional question, where is the console output itself for the shader compilation ?

Thanks!

**Comment by pragma37** [comment]
Changes in the shader file should be detected and reloaded automatically.
Could you share a reproduction file so I can test it?

> where is the console output itself for the shader compilation ?

There's no console output. Any error/warning should show in the material panel.

**Comment by panthavma** [comment, marked as answer]
Hello, here is a reproduction file.

OS : Windows 10
Blender : 3.0.1
Malt : Preview 1.0

[2022-03-10-shaderreload.zip](https://github.com/bnpr/Malt/files/8222156/2022-03-10-shaderreload.zip)

When I was recreating it I used the wrong pipeline by mistake (CustomPipeline.py instead of MiniPipeline) and it seemed to reload fine, so it might be a problem with MiniPipeline itself

Another bug (?) I found with that CustomPipeline.py was that it didn't seem to start correctly when opened, until I made a change to the pipeline. I'll try making a repro case for that too later.

**Reply by pragma37** [reply]
There was a bug caused by the Nodetree materials from the default NPR_Pipeline left in your scene.
It's fixed in the latest development version. 👍 

> Another bug (?) I found with that CustomPipeline.py was that it didn't seem to start correctly when opened, until I made a change to the pipeline. I'll try making a repro case for that too later.

~~Yes, sorry. Some of the documentation is outdated. CustomPipeline is outdated, use the MiniPipeline instead.~~
Edit: Seems like the docs for custom pipelines are up to date?

**Reply by panthavma** [reply]
Nice! Thanks for the answer!
